packages/sagepay-integration/src/sage-pay-payment-strategy.ts

```diff
@@ -88,24 +80,21 @@ export default class SagePayPaymentStrategy extends CreditCardPaymentStrategy {
         });
     }
 
-    finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
-        const state = this._store.getState();
-        const order = state.order.getOrder();
+    async finalize(options?: PaymentRequestOptions): Promise<void> {
+        const state = this.paymentIntegrationService.getState();
+        const order = state.getOrder();
 
-        if (order && state.payment.getPaymentStatus() === paymentStatusTypes.FINALIZE) {
-            return this._store.dispatch(
-                this._orderActionCreator.finalizeOrder(order.orderId, options),
-            );
+        if (order && state.getPaymentStatus() === PaymentStatusTypes.FINALIZE) {
+            await this.paymentIntegrationService.finalizeOrder(options);
```

Comment:
By looking for prev implementation I see that `super.finalize()` should not be called for current condition. So it might be better to update this code with:
```suggestion
            return this.paymentIntegrationService.finalizeOrder(options);
```

Reply:
Looks like I can't do it, I get error 'finalize' in 'SagePayPaymentStrategy' is not assignable to the same property in base type 'CreditCardPaymentStrategy', and it has to return Promise<void>